buildSrc/src/main/groovy/com/sun/javafx/gradle/CCTask.groovy

```diff
@@ -23,15 +23,16 @@
  * questions.
  */
 
+import org.gradle.api.tasks.Input
 import org.gradle.api.tasks.InputDirectory
 import org.gradle.api.tasks.Optional
 
 class CCTask extends NativeCompileTask {
-    String compiler;
-    @Optional List<String> linkerOptions = new ArrayList<String>();
+    @Input String compiler;
+    @Optional @Input List<String> linkerOptions = new ArrayList<String>();
     @Optional @InputDirectory File headers;
-    @Optional Closure eachOutputFile; // will be given a File and must return a File
-    @Optional boolean exe = false;
+    @Optional @Input Closure eachOutputFile; // will be given a File and must return a File
```

Comment:
Is this field actually used anywhere?

As this field is not initialized + there is the `@Optional` annotation, Gradle will not verify this is fine. Gradle docs is specifying that:
1. `@Input` can be used on Closure if it is a provider of `File` (which means no param in closure).
2. `@Input` can be used for any `Serializable` which `Closure` actually is.

It's not obvious if Gradle will correctly handle `Closure` taking argument.

Reply:
I'll take a look, but I will likely want to do any cleanup of this in a follow-on issue. As you note the `eachOutputFile` property is unused...as is the `exe` property. They were formerly used by one of the `javapackager` tasks, which has since been removed.